src/cmake-tools.ts

```diff
@@ -401,16 +444,27 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
    */
   get allTargetName() { return this._allTargetName(); }
   private async _allTargetName(): Promise<string> {
-    const drv = await this._cmakeDriver;
-    return drv.allTargetName;
+    const drv = await this.getCMakeDriverInstance();
+    if (drv) {
+      return drv.allTargetName;
+    } else {
+      return "";
+    }
   }
 
   /**
    * Implementation of `cmake.build`
    */
   async build(target_?: string): Promise<number> {
     // First, reconfigure if necessary
-    const drv = await this._cmakeDriver;
+    const drv = await this.getCMakeDriverInstance();
+    if (!drv) {
+      log.debug('Build of project is not possible, because driver is not ready.');
+      vscode.window.showErrorMessage('Unable to build, try to run configure before build.');
+
+      return -1;
```

Comment:
I think this should be returned as a `Promise`:
```ts
return Promise.resolve(-1);
```

Reply:
Style guide question